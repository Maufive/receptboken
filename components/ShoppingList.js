import React, { Component } from "react";
import axios from "axios";
import Link from "next/link";
import { Loading } from "./Loading";
import { List, ListContainer } from "./styles/ShoppinglistStyles";

class ShoppingList extends Component {
	state = {
		shoppinglists: null
	};

	componentDidMount() {
		if (this.props.user) {
			this.getShoppingLists();
		}
	}

	getShoppingLists = () => {
		axios.defaults.headers.common["Authorization"] =
			"Bearer " + localStorage.getItem("jwtToken");
		axios
			.get(`http://localhost:7777/lists/user/${this.props.user._id}`)
			.then(response => this.setState({ shoppinglists: response.data }))
			.catch(error => console.log(error));
	};

	render() {
		const { shoppinglists } = this.state;
		if (!shoppinglists || shoppinglists.length < 1) return <Loading />;
		return (
			<ListContainer>
				<h1>Sparade ink√∂pslistor:</h1>
				<List>
					{shoppinglists.map(list => (
						<Link
							href={{
								pathname: "/inkopslista",
								query: {
									id: list._id
								}
							}}
						>
							<a>
								<li key={list._id}>{list.title}</li>
							</a>
						</Link>
					))}
				</List>
			</ListContainer>
		);
	}
}

export default ShoppingList;
