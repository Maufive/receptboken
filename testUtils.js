export const fakeRecept = {
	_id: "1234",
	author: "5678436374re67fwuf76",
	created: "Idag",
	description: ["Steg1", "Steg2", "Steg3"],
	id: "1337",
	ingredients: [
		{
			input: "Lök",
			numberOfUnits: "2",
			units: "dl"
		}
	],
	photo: "asdf123.jpg",
	reviews: [{ score: 1 }],
	servings: "2",
	tags: ["Mat", "Gott"],
	timeRequired: "20 min",
	title: "Mitt jättegoda recept"
};

export const fakeUser = {
	_id: "1234",
	username: "fakeuser@gmail.com",
	password: "1234",
	hearts: [],
	fname: "Niklas",
	lname: "Albinsson",
	description: "Hej jag heter Niklas",
	photo: "http://fakephoto.jpg"
};

export const fakeShoppinglist = [
	{
		_id: "34567876543452176",
		title: "My Shoppinglist",
		created: "12/3 2019",
		author: "61386712376576d7q6ewd65",
		list: [
			{
				input: "Lök",
				numberOfUnits: "5",
				units: "dl"
			},
			{
				input: "Tomater",
				numberOfUnits: "10",
				units: "st"
			}
		]
	}
];
