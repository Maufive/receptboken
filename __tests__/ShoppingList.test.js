import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import ShoppingList from "../components/ShoppingList";
import { fakeShoppinglist } from "../testUtils";

describe("<ShoppingList />", () => {
	let wrapper;
	beforeEach(() => (wrapper = shallow(<ShoppingList />)));

	it("renders without errors without state", () => {
		expect(wrapper.find("p").text()).toBe("Du har inga inkÃ¶pslistor...");
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	it("renders out the shoppinglists when state is set", () => {
		wrapper.setState({ shoppinglists: fakeShoppinglist });
		expect(wrapper.find("ShoppinglistStyles__List").children().length).toBe(1);
		console.log(wrapper.find("ShoppinglistStyles__List").children().length);
	});
});
