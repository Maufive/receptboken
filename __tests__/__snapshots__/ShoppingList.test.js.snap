// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShoppingList /> renders without errors without state 1`] = `
<div>
  <ShoppinglistStyles__ListContainer>
    <h1>
      Sparade inköpslistor:
    </h1>
    <ShoppinglistStyles__List>
      <li>
        <p>
          Du har inga inköpslistor...
        </p>
      </li>
    </ShoppinglistStyles__List>
  </ShoppinglistStyles__ListContainer>
</div>
`;
